crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -26,14 +32,35 @@ pub(crate) trait EthSigner: Send + Sync {
         request: TypedTransactionRequest,
         address: &Address,
     ) -> Result<TransactionSigned>;
+
+    /// Encodes and signs the typed data according EIP-712. Payload must implement Eip712 trait.
+    fn sign_typed_data(&self, address: Address, payload: &TypedData) -> Result<Signature>;
 }
 
 /// Holds developer keys
 pub(crate) struct DevSigner {
     addresses: Vec<Address>,
     accounts: HashMap<Address, SecretKey>,
+    sign_engine: Secp256k1<All>,
```

Comment:
I think Matt meant using the global `SECP256K1` like we do here https://github.com/paradigmxyz/reth/blob/a0f195590c5b0bc5d7b32a9f449db80aac7018e7/crates/primitives/src/transaction/util.rs#L7?